src/main/java/javaslang/collection/Vector.java

```diff
@@ -1342,9 +1342,9 @@ private Object readResolve() {
     public boolean equals(Object o) {
         if (o == this) {
             return true;
-        } else if (o instanceof Vector) {
+        } else if (o instanceof IndexedSeq) {
```

Comment:
Maybe you're right and I should rewrite <code>StringTest.assertThat()</code> method if we don't need in this functionality outside of tests